src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -184,9 +190,31 @@ public static void ConfigureConnectionStringManifestPublisher(IResourceBuilder<I
     public static ParameterResource CreateGeneratedParameter(
         IDistributedApplicationBuilder builder, string name, bool secret, GenerateParameterDefault parameterDefault)
     {
-        var parameterResource = new ParameterResource(name, parameterDefault => GetParameterValue(builder.Configuration, name, parameterDefault), secret);
+        var parameterResource = CreateGeneratedParameter(builder.Configuration, name, secret, parameterDefault);
 
-        parameterResource.Default = parameterDefault;
+        if (builder.ExecutionContext.IsRunMode && parameterResource.Default is not null)
+        {
+            parameterResource.Default = new UserSecretsParameterDefault(builder.AppHostAssembly, builder.Environment.ApplicationName, name, parameterResource.Default);
+        }
+
+        return parameterResource;
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="ParameterResource"/> that has a generated value using the <paramref name="parameterDefault"/>.
+    /// </summary>
+    /// <param name="configuration">The application configuration.</param>
+    /// <param name="name">Name of parameter resource.</param>
+    /// <param name="secret">Flag indicating whether the parameter should be regarded as secret.</param>
+    /// <param name="parameterDefault">The <see cref="GenerateParameterDefault"/> that describes how the parameter's value should be generated.</param>
+    /// <returns>The created <see cref="ParameterResource"/>.</returns>
+    public static ParameterResource CreateGeneratedParameter(
+        IConfiguration configuration, string name, bool secret, GenerateParameterDefault parameterDefault)
```

Comment:
This is kind of surprising for the overload that takes an `IDistributedApplicationBuilder` to save the value to user secrets, but the overload that just takes an `IConfiguration` doesn't.

Should/can we just make this private until we need to expose it?

Reply:
These aren't new methods though. Previously the overload that accepted the builder just forwarded to the one that takes configuration. This was one of the reasons why I avoided doing it here originally as this method chain is all public and adding new optional parameters or overloads wasn't super appealing.

Reply:
ok let me take another look lol

Reply:
OK I removed the overload that takes `IConfiguration`